pkg/lang/ir/v1/system.go

```diff
@@ -46,6 +47,82 @@ func (g generalGraph) compileUbuntuAPT(root llb.State) llb.State {
 	return root
 }
 
+func (g generalGraph) copyAptSignature(root llb.State, name string, url string) (llb.State, string) {
+	var fileName = fmt.Sprintf("%s.asc", name)
+	var filePath = "/etc/apt/keyrings"
+
+	base := llb.Image(builderImage)
+	builder := base.
+		Run(llb.Shlex(fmt.Sprintf("sh -c \"curl %s >> %s\"", url, fileName)),
+			llb.WithCustomName("[internal] downloading apt-source signature in base image")).Root()
+
+	aptSign := root.
+		File(llb.Mkdir(filePath, 0755, llb.WithParents(true)),
+			llb.WithCustomName("[internal] setting target apt-source signature folder")).
+		File(llb.Copy(builder, fileName, filePath+fileName),
+			llb.WithCustomName("[internal] copy signature from builder"))
+
+	return aptSign, filePath + fileName
+}
+
+func (g generalGraph) configRSrc(root llb.State, aptConfig ir.AptConfig, sign string) (llb.State, string) {
```

Comment:
Could you please add some comments to this func?

Reply:
Added comments